### PR TITLE
fix: proper `runtimeConfig` and `publicRuntimeConfig` declaration 

### PR DESCRIPTION
**Pull request by AndreyYolkin**
### 🔗 Linked issue
Resolves https://github.com/nuxt/bridge/issues/948

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This PR backports schema generation approach from https://github.com/nuxt/nuxt/blob/4b2608c74d80551eee857ab74ab470da839894f9/packages/nuxt/src/core/templates.ts#L132, which introduces next fixes/features:

1. Runtime Config (and its public part) now has more accurate types, generated with Untyped
2. `$config` in Vue templates is declared with the same generated type

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### Before:
image
image

### After:
image

(Untyped has some struggles with array types, not Nuxt itself issue):
image

### CODE REVIEW

**Review by danielroe — COMMENTED** [review]
This is great! I wonder if we ned to do this in a separate template so we can do it conditionally on whether `nitro` is enabled or not as I think the runtimeConfig has a different format if it's disabled.

**Comment by AndreyYolkin** [comment]
Wise enough. I'll take another round of refactoring to ensure it has support for nitro-less build

**Comment by danielroe** [comment]
(converting to draft just for more obvious triage - feel free to mark ready any time you feel like it)

File: packages/bridge/src/type-templates.ts [hunk 1]
```diff
@@ -52,31 +55,98 @@ export const middlewareTypeTemplate = {
...
+
+    const bridgeSchemaConfig = [
+      'declare module \'@nuxt/bridge-schema\' {',
+      '  interface NuxtApp {',
+      '    $config: RuntimeConfig',
+      '  }',
+      '  interface NuxtAppCompat {',
+      '    $config: RuntimeConfig',
+      '  }',
+      '  interface NuxtRuntimeConfig extends RuntimeConfig {}',
```
Comment (AndreyYolkin) [inline packages/bridge/src/type-templates.ts:132]: I don't really know about this: we can use 
```ts
const { $config } = useNuxtApp()

const { nuxt2Context: { $config } } = useNuxtApp()
```

but displayed types will be different

**Review by wattanx — APPROVED** [review]
Nice! Looks good to me!

**Comment by danielroe** [comment]
Beautiful work! In future we likely need to add a type test and confirm that `useRuntimeConfig()` works properly (right now it seems to be shadowed by Nitro auto-import).

**Pull request merged**